controllers/cronset_controller.go

```diff
@@ -66,6 +66,19 @@ func (r *CronSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	var nodeSelector map[string]string
+	if obj.Spec.CronJobTemplate.Spec.JobTemplate.Spec.Template.Spec.NodeSelector != nil {
+		nodeSelector = obj.Spec.CronJobTemplate.Spec.JobTemplate.Spec.Template.Spec.NodeSelector
+	}
+	nodes := &corev1.NodeList{}
+	if err := r.Client.List(ctx, nodes, client.MatchingLabels(nodeSelector)); err != nil && errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	}
+	nodeMap := make(map[string]bool)
```

Comment:
how is your plan to use this nodeMap at the following step?

Reply:
We need to remove unnecessary cronjobs and then create or update cronjobs based on the node list.

I want to iterate through the list of cronjobs and if the node name of a child Pod is not in the node list, I want to remove that cronjob.
At this point, I think I might get better performance by converting the node list to a map.

Note that for now, I only plan to use node maps for removing cronjobs, and will not use them to create or update cronjobs during node list iterations.
Maps seem to have a higher iteration cost than slices.

Reply:
the nodemap that is may better to make at the next step what you tell me. It makes me a little confused, cuz the PR is about getting the list nodes. but I understand what you want due to you described specifically. thanks. :)